client-sdk/ts-web/rt/playground/src/consensus.js

```diff
@@ -25,6 +25,51 @@ const nic = new oasis.client.NodeInternal('http://localhost:42280');
 const accountsWrapper = new oasisRT.accounts.Wrapper(CONSENSUS_RT_ID);
 const consensusWrapper = new oasisRT.consensusAccounts.Wrapper(CONSENSUS_RT_ID);
 
+/**
+ * Await this so that this function can get the starting block before you go on.
+ * Callback should return false to stop.
+ * Returns a promise for when polling stops, wrapped in an object to prevent automatic resolving.
+ * @param {(e: oasis.types.RuntimeClientEvent) => boolean} cb
+ */
+async function pollEvents(cb) {
+    const startBlock = await nic.runtimeClientGetBlock({
+        runtime_id: CONSENSUS_RT_ID,
+        round: oasis.runtime.CLIENT_ROUND_LATEST,
+    });
+    const done = (async () => {
+        let nextRound = BigInt(startBlock.header.round) + 1n;
+        poll_blocks: while (true) {
+            // Local testnet runs faster. Use ~6_000 for Oasis testnet and mainnet.
+            await delay(1_000);
+            let events;
+            try {
+                events = await nic.runtimeClientGetEvents({
+                    runtime_id: CONSENSUS_RT_ID,
+                    round: nextRound,
+                });
+            } catch (e) {
+                if (
+                    e.oasisModule === oasis.roothash.MODULE_NAME &&
+                    e.oasisCode === oasis.roothash.ERR_NOT_FOUND_CODE
+                ) {
+                    // Block doesn't exist yet. Wait and fetch again.
+                    continue;
+                }
+                throw e;
+            }
+            console.log('polled block', nextRound);
+            if (events) {
+                for (const e of events) {
+                    if (!cb(e)) break poll_blocks;
```

Comment:
this works for a test, but there's no mechanism to cancel this if the tx submission fails. we probably need more weird cancellation flags and stuff